pkg/arvo/age/clock.hoon

```diff
@@ -0,0 +1,64 @@
+/+  *server, default-agent, verb
+/=  tile-js
+  /^  octs
+  /;  as-octs:mimes:html
+  /:  /===/app/clock/js/tile
+  /|  /js/
+      /~  ~
+  ==
+=,  format
+::
+%+  verb  &
+^-  agent:mall
+%+  http-handler
+  %-  require-authorization:app
+  |=  =inbound-request:eyre
+  ^-  response:http-handler
+  =/  request-line  (parse-request-line url.request.inbound-request)
+  =/  back-path  (flop site.request-line)
+  =/  name=@t
+    =/  back-path  (flop site.request-line)
+    ?~  back-path
+      ''
+    i.back-path
+  ::
+  ?~  back-path
+    not-found:gen
+  ?:  =(name 'tile')
+    (js-response:gen tile-js)
+  not-found:gen
```

Comment:
```
  ?+  (flop site.request-line)  not-found:gen
    [%tile *]  (js-response:gen tile-js)
  ==
```

also, am I right to understand `server` is implicitly exporting a three-letter named _core_ with a bunch of helper functions? that was kind of hard to follow, I realize we're moving away from three-letter names for local variables but imo we shouldn't move towards them for other things

also also this seems like kind of a lot of logic for "on `GET /~clock/js/tile.js` serve `/app/clock/js/tile.js`"(or well also on `POST` or `HEAD` or `OPTIONS` :/ ), this seems useful config to keep bundled somewhere near the app but I'm not sure "custom handler" is the place. Unless this code is trying to gesture at like, \~prefetching\~ JSON into the initial JS fetch?

Reply:
First, compare what's on master now: https://github.com/urbit/urbit/blob/a480e3aafd483d21aa2d589c1363ad2a502f32ef/pkg/arvo/app/clock.hoon

Second, yeah, I'm not a big fan of using `*server` like that.  I feel like "app" and "gen" aren't too bad for application and generator, but only if you know where to look for them.  But I'm converting a bunch of stuff to static gall right now, so I've declared those sorts of changes to be out of scope for now.

Third, how about what I just pushed.  Except for `/+  http-handler`, `%-  http-handler` and `+on-poke`, the rest is just registering the tile with `%launch` (which could abstracted into another library if desired).

Reply:
Okay yup, completely missed that this was a copy of an existing file ty

I am alas only able to compare to "back in my day", where requests for static assets weren't being misrouted to apps in the first place. (I see there's also login logic there, so not completely static, but it seems like a lot of work to go for a boolean auth/noauth flag)

Reply:
There's a generator path in eyre that should probably be used for this when it's static.  The only part that "needs" an app is registering the generator (the `%bound` request) and registering the tile with launch.  And both of those are stateless so could be done by a more powerful version of generators (possibly imps will morph into this).

Reply:
This looks nice to me as a straightforward port of existing functionality.